Services/Catalog/Catalog.DAL/Repositories/Interfaces/IGenericRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Catalog.DAL.Entities;
+
+namespace Catalog.DAL.Repositories.Interfaces
+{
+    public interface IGenericRepository<TEntity> where TEntity : Entity
+    {
+        Task<List<TEntity>> Find();
+        Task<TEntity> FindById(Guid id);
+        Task<TEntity> FindLast();
+        void InsertOrUpdate(TEntity entity);
```

Comment:
better be changed to asynchronous versions

Reply:
Add, update and remove don't have async versions because they only change state in which entity is tracked